util/progress/progressui/display.go

```diff
@@ -21,11 +21,16 @@ import (
 	"golang.org/x/time/rate"
 )
 
-func DisplaySolveStatus(ctx context.Context, phase string, c console.Console, w io.Writer, ch chan *client.SolveStatus) ([]client.VertexWarning, error) {
+func DisplaySolveStatus(ctx context.Context, phase string, fndesc func() (string, string), c console.Console, w io.Writer, ch chan *client.SolveStatus) ([]client.VertexWarning, error) {
```

Comment:
nit: since this is a public function, can we create a public type for this?

Is there a reason this is a function btw, instead of just a struct pointer?

Reply:
Switched to variadic opt

> Is there a reason this is a function btw, instead of just a struct pointer?

Easier to handle when it needs inline logic: https://github.com/docker/buildx/pull/1177/files#diff-3f9c21dce82a82fae6d5ca096e77e5ddcf0eccd7051f815b452ec26fbfeb314bR114-R123

Reply:
Variadic opt looks good to me :+1: 

I think it would still be good to use a struct instead, since we always call that function anyways (so it's not saving an expensive operation or similar), and there are no parameters passed from inside `DisplaySolveStatus`.